lib/fog/openstack/models/compute/server.rb

```diff
@@ -117,11 +117,19 @@ def reload
         # returns all ip_addresses for a given instance
         # this includes both the fixed ip(s) and the floating ip(s)
         def ip_addresses
-          addresses.values.flatten.map{|x| x['addr']}
+          addresses ? addresses.values.flatten.collect { |x| x['addr'] } : []
         end
 
         def floating_ip_addresses
-          all_floating=addresses.values.flatten.select{ |data| data["OS-EXT-IPS:type"]=="floating" }.map{|addr| addr["addr"] }
+          all_floating = if addresses
+                           addresses.values.flatten.select { |data|
+                             data["OS-EXT-IPS:type"] == "floating"
+                           }.collect { |addr|
```

Comment:
-1
Hound is right here, the block is hard to read, please use do...end over multiple lines.
Also you might have to use a proper if conditional block.